datadog-setup.php

```diff
@@ -1574,7 +1574,7 @@ function ini_values($binary)
 
     if ($found[EXTENSION_DIR] == "") {
         $found[EXTENSION_DIR] = dirname(PHP_BINARY);
-    } elseif ($found[EXTENSION_DIR][0] != "/" && (!IS_WINDOWS || !preg_match('~^([A-Z]:|\\\\)\\\\~i', $found[EXTENSION_DIR]))) {
+    } elseif ($found[EXTENSION_DIR][0] != "/" && (!IS_WINDOWS || !preg_match('~^([A-Z]:[\\\\/]|\\\\{2})~i', $found[EXTENSION_DIR]))) {
```

Comment:
Shouldn't it be `~^([A-Z]:[\\/]|\\{2})~i`?

- `[\\\\/]` changed to `[\\/]`. No need to repeat `\\` twice in the list of allowed chars
- `\\\\{2}` changed to `\\{2}`. Otherwise we must match `\` 3 times

Reply:
`'\\'` in a string will be equivalent to a single \.
However in regex, you need to escape all metacharacters as well. Thus doubling the \\.